server/util/grpc_client/grpc_client.go

```diff
@@ -6,16 +6,28 @@ import (
 	"net/url"
 	"time"
 
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/rpc/interceptors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/google"
 	"google.golang.org/grpc/keepalive"
 )
 
-// DialTarget handles some of the logic around detecting the correct GRPC
+// DialSimple handles some of the logic around detecting the correct GRPC
 // connection type and applying relevant options when connecting.
-func DialTarget(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
-	dialOptions := CommonGRPCClientOptions()
+//
+// NOTE: Clients within BuildBuddy servers (i.e. executor) should use the DialInterservice
+// variant with an env argument.
+func DialSimple(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
+	return DialInterservice(nil, target, extraOptions...)
+}
+
+// DialInterservice is similar to DialSimple, but it adds additional interceptors
+// (such as client identity) based on the specified environment.
+//
+// Outside of BuildBuddy servers, DialSimple may be used instead.
+func DialInterservice(env environment.Env, target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
nit: can you keep comments to < 80 chars and add some detail here about cases where each of these is appropriate? Want to make it crystal clear which one should be used for new cases (can we replace 'may' with 'should' and clarify?)

Reply:
done